src/main/java/com/conveyal/gtfs/graphql/fetchers/JDBCFetcher.java

```diff
@@ -70,6 +76,87 @@
     private final String sortField;
     private final boolean autoLimit;
 
+    /**
+     * Get a new MappedBatchLoaderWithContext batch loader that is able to batch multiple queries and cache the results.
+     */
+    public static DataLoader<JdbcQuery, List<Map<String, Object>>> newJdbcDataLoader() {
+        return DataLoader.newMappedDataLoader(
+            (jdbcQueries, context) -> CompletableFuture.supplyAsync(() -> getSqlQueryResults(jdbcQueries, context))
+        );
+    }
+
+    /**
+     * Method where batched sql queries are all ran at once.
+     */
+    private static Map<JdbcQuery, List<Map<String, Object>>> getSqlQueryResults(
+        Set<JdbcQuery> jdbcQueries,
+        BatchLoaderEnvironment context
+    ) {
+        Map<JdbcQuery, List<Map<String, Object>>> allResults = new HashMap<>();
+        Map<Object, Object> keyContexts = context.getKeyContexts();
+        for (JdbcQuery jdbcQuery : jdbcQueries) {
```

Comment:
What is the goal of batching SQL queries together in this way? It seems to just run them all sequentially on separate connections.

Reply:
This does seem like it could definitely be made more efficient by creating a connection at the beginning of the graphQL request, reusing that connection in the DataFetchingEnvironment and finally closing the connection after the execution has completed.

Reply:
Actually, doesn't the datasource manage connections via a pool?

Reply:
@evansiroky clarified in conversation that this construct that is handed a lot of queries and executes them all at once is just part of the design of the data loader mechanism. It gives you a hook where you can see a large number of outstanding queries at once, and attempt to perform some optimizations on them before they are run (and gives you control over how they are run). In the end you need to associate sets of responses with each incoming unique query, the exact optimizations are up to you.

Reply:
@evansiroky about the connections, yes we're using a connection pool. So the amortized cost of getting a connection is very low, since they're being reused. We're not creating new TCP connections all the time. There is some cost to releasing and re-using the connection though. I wasn't implying that we should grab and reuse a single database connection for all the queries (though that does seem like a reasonable optimization and should be harmless).

I was just trying to get my head around why there was an extension point where we receive a bunch of requests and then re-issue them all one after the other. It clearly allows optimizations to be performed on the whole batch of queries, but since certain types of optimizations were not being performed I didn't realize the full purpose of this design.